pkg/parsers/github/steps.go

```diff
@@ -101,18 +103,41 @@ func parseActionHeader(header string) (string, string, models.VersionType) {
 	return actionName, version, versionType
 }
 
-func parseActionInput(with map[string]any) *[]models.Parameter {
+func parseActionInput(with *githubModels.With) *[]models.Parameter {
 	if with == nil {
 		return nil
 	}
 
-	parsedInputs := utils.MapToSlice(with, parseActionInputItem)
-	return &parsedInputs
+	parameters := make([]models.Parameter, 0)
+	index := 0
+
+	for key, value := range with.Inputs {
+		name := key
+		parameter := models.Parameter{
+			Name:          &name,
+			Value:         value,
+			FileReference: calcParameterFileReference(index, key, value, with.FileReference),
+		}
+		parameters = append(parameters, parameter)
+		index += 1
+	}
+
+	return &parameters
 }
 
-func parseActionInputItem(k string, val any) models.Parameter {
-	return models.Parameter{
-		Name:  &k,
-		Value: val,
+func calcParameterFileReference(index int, key string, val any, fileReference *models.FileReference) *models.FileReference {
+	if fileReference == nil {
+		return nil
+	}
+
+	return &models.FileReference{
+		StartRef: &models.FileLocation{
+			Line:   fileReference.StartRef.Line + index + 1,
+			Column: fileReference.StartRef.Column + 2, // for the tab after the inputs
+		},
+		EndRef: &models.FileLocation{
+			Line:   fileReference.StartRef.Line + index + 1 + strings.Count(fmt.Sprint(val), "\n"),
```

Comment:
Since you're using the index and the current amount of linebreaks, if you don't consider the previous inputs with the line breaks when you calculate the current input, you'll end up with an input with a start line that is lower than the previous input's end line